src/parser/bison_parser.y

```diff
@@ -730,7 +732,10 @@ literal_list:
 expr_alias:
 		expr opt_alias {
 			$$ = $1;
-			$$->alias = $2;
+			if ($2) {
+				$$->alias = strdup($2->name);
+				delete $2;
```

Comment:
On second thought the grammar is now screwed. Supporting both table and column aliases requires a deeper separation of concerns.
As we can see here, an alias for `expr` re-uses the general purpose alias rule, discarding possible columns.
As a result something like `select a + 1 as b(spam, eggs) from B;` would be successfully parsed in the current state. I'll attempt at a better solution.

Reply:
Clarification: done. Shouldn't have reused that particular grammar rule in the first place.